lib/url.js

```diff
@@ -353,6 +353,13 @@ function urlFormat(obj) {
   // this way, you can call url_format() on strings
   // to clean up potentially wonky urls.
   if (typeof obj === 'string') obj = urlParse(obj);
+
+  if (typeof obj !== 'object')
```

Comment:
Optional: If `urlParse()` guarantees to return an object, then you could avoid these two checks by making them `else if`s. If you opt for a simpler error message, you could combine the two into one `else`.